apis/dns/v1alpha1/managed_zone_types.go

```diff
@@ -0,0 +1,137 @@
+	//   "public"
+	//   "private"
+	// +optional
+	// +immutable
+	// +kubebuilder:validation:Enum=public;private
+	Visibility *string `json:"visibility,omitempty"`
+
+	// TODO(danielinclouds): support CloudLoggingConfig parameters
+	// TODO(danielinclouds): support DnssecConfig parameters
+	// TODO(danielinclouds): support ForwardingConfig parameters
+	// TODO(danielinclouds): support NameServerSet parameters
+	// TODO(danielinclouds): support PeeringConfig parameters
+	// TODO(danielinclouds): support ReverseLookupConfig parameters
+	// TODO(danielinclouds): support ServiceDirectoryConfig parameters
+}
+
+// ManagedZonePrivateVisibilityConfig the set of Virtual Private Cloud resources
+// that the zone is visible from
+type ManagedZonePrivateVisibilityConfig struct {
+
+	// Networks: The list of VPC networks that can see this zone.
+	Networks []*ManagedZonePrivateVisibilityConfigNetwork `json:"networks"`
+}
+
+// ManagedZonePrivateVisibilityConfigNetwork is a list of VPC networks
+type ManagedZonePrivateVisibilityConfigNetwork struct {
+
+	// NetworkUrl: The fully qualified URL of the VPC network to bind to.
+	// Format this URL like
+	// https://www.googleapis.com/compute/v1/projects/{project}/global/networks/{network}
+	NetworkURL *string `json:"networkUrl"`
+}
+
+// ManagedZoneObservation is used to show the observed state of the ManagedZone
+type ManagedZoneObservation struct {
+
+	// CreationTime: The time that this resource was created on the server.
+	// This is in RFC3339 text format. Output only.
+	CreationTime string `json:"creationTime,omitempty"`
+
+	// Id: Unique identifier for the resource; defined by the server (output only)
+	ID uint64 `json:"id,omitempty"`
+
+	// NameServers: Delegate your managed_zone to these virtual name
+	// servers; defined by the server (output only)
+	NameServers []string `json:"nameServers,omitempty"`
+}
+
+// ManagedZoneSpec defines the desired state of a ManagedZone.
+type ManagedZoneSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+	ForProvider       ManagedZoneParameters `json:"forProvider"`
+}
+
+// ManagedZoneStatus represents the observed state of a ManagedZone.
+type ManagedZoneStatus struct {
+	xpv1.ResourceStatus `json:",inline"`
+	AtProvider          ManagedZoneObservation `json:"atProvider,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// ManagedZone is a managed resource that represents a Managed Zone in Cloud DNS
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="DNS NAME",type="string",JSONPath=".spec.forProvider.dnsName"
+// +kubebuilder:printcolumn:name="VISIBILITY",type="string",JSONPath=".spec.forProvider.visibility"
+// +kubebuilder:printcolumn:name="DESCRIPTION",type="string",JSONPath=".spec.forProvider.description"
```

Comment:
I think the `DESCRIPTION` printer column is not necessary. It isn't an important field that would be nice to understand when `kubectl get` the resource. Seeing it in `kubectl describe` is enough.

Other problem is that this field can be a long string and this wouldn't be a good experience trying to `kubectl get`, as it would break lines in the terminal

Reply:
fixed